src/compiler/checker.ts

```diff
@@ -6093,6 +6107,16 @@ namespace ts {
                                 }
                             }
                         }
+                        if (signatures.some(s => s.declaration && getModifierFlags(s.declaration) & ModifierFlags.Private)) {
+                            return [setTextRange(createProperty(

```

Comment:
I don't think 
```ts
class Foo {
  private constructor;
}
```
is the same as
```ts

class Foo {
  private constructor();
}
```
?

Specifically, the first is actually a parse error? Is there a test of this?

Reply:
Is this code JS-specific? Trying to figure out why it didn’t exist before now.

Reply:
Fixed